packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -433,6 +467,174 @@ export class GraphQLApi extends Construct {
     });
   }
 
+  /**
+   * Adds an IAM policy statement associated with this GraphQLApi to an IAM
+   * principal's policy.
+   *
+   * @param grantee The principal (no-op if undefined)
+   * @param resources The set of resources to allow (i.e. ...:[region]:[accountId]:apis/GraphQLId/...)
+   * @param actions The actions that should be granted to the principal (i.e. appsync:graphql )
+   */
+  public grant(grantee: IGrantable, resources: IamResources[], ...actions: string[]): Grant {
+    return Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [...resources.map(resource => Lazy.stringValue({
+        produce: () => {
+          let resourceTypes;
+          if (resource.custom) {
+            resourceTypes = `${resource.custom}`;
+          } else if (resource.type && resource.field) {
+            resourceTypes = `types/${resource.type}/fields/${resource.field}`;
+          } else if (resource.type) {
+            resourceTypes = `types/${resource.type}/*`;
+          } else {
+            resourceTypes = '*';
+          }
```

Comment:
there is cleaner way to do this with ternary operators, but don't know if the logic would be harder to follow. 

What would probably be better would be a function I think that creates the `resourceName` field I think..